### ISSUE TITLE
Create runner addin for VS Code when extensions become available

### ISSUE THREAD

**Issue by LokiMidgard** [statiqdev/Statiq#91]
If you use `--preview` and the console doesn't support ReadKey(), like in visual studio code. The process just ends after showing `Hit any key to exit`. Most likely because of an exception even through I don't see it on the console.

Should we just wrap the code in an `try` and in the `catch` we tell the user that any keys are not supported?

**Comment by daveaglick** [comment]
Hadn't even considered that, thanks. Shouldn't be hard to resolve - I'll rig something up within the next day or two.

**Comment by daveaglick** [comment]
How are you running Wyam in VS Code? Through a custom task?

**Comment by LokiMidgard** [comment]
Yes. currently I still testing what works best and what is missing.

I will create some more issues for that, if I have time.

**Comment by daveaglick** [comment]
Maybe I'm misunderstanding this one. I've created a task in Visual Studio Code that looks like this:

```
{
    "version": "0.1.0",
    "command": "e:\\Code\\Wyam\\Wyam\\bin\\Debug\\wyam.exe",
    "isShellCommand": true,
    "showOutput": "always",
    "args": ["--preview"]
}
```

It runs Wyam as expected, but the output window in VS Code is just that, it displays the output. As far as I can tell, it's not actually a shell. It's just redirecting the console out from whatever task was run to the output window. Because of that, there's no opportunity to provide anything on the console input stream, single keys or otherwise.

I do think the kind of behavior you're going for (triggering a Wyam build and watch right from the editor) would be great. I just don't think the current task/build integration in VS Code is going to provide the interactivity needed for starting and stopping the preview server. Once VS Code gets their addin model sorted out, I would love to re-address this and see if we can get some kind of plugin for always-on Wyam watching, building, and preview (similar to how OmniSharp runs in the background).

Let me know if I've misunderstood the problem.

**Comment by LokiMidgard** [comment]
That's exactly what I do. I use `"isShellCommand": false` but I don't think that changes something. Also I use `"isWatching"`
For the interactivity I need to stop it I use `>Tasks:Terminate Running Task`. Most likely not a really ordered shutdown, but it worked for me.

Normally I just have it running while working so my output constantly get updated. Until I miss a semicolon on the wrong place and the process completely stops and I have to restart it.

**Comment by daveaglick** [comment]
Okay good, so we're on the same page. I'm going to go ahead and rename/repurpose this issue to creating a VS Code addin once their extensibility story gets straightened out. This is probably also related to #21 since both deal with better background generation.

**Comment by daveaglick** [comment]
Related to #100
